### PR TITLE
MCMC plots: properly rescale injection parameters

### LINKED ISSUE #166: MCMC walker plots: injection parameter not properly scaled
When doing `mcmc.run(walker_plot_args={"plot_det_stat": True, "injection_parameters": inj})` on a `MCMCTransientSearch` object, the injected start/duration parameters are not rescaled and hence the plot looks completely off:

transient_search_walkers

The `transform_dictionary` is set up properly and `MCMCSearch._plot_walkers()` (which is inherited unmodified) does have a `_scale_samples()` call, so I'm not sure yet where the crucial info gets lost - probably a parameter name inconsistency somewhere.

### PR DESCRIPTION
**Pull request by dbkeitel**
Let's use this as a starting point to fix #166. So far only amends one example to expose the issue in the walker plot:

PyFstat_example_fully_coherent_MCMC_search_walkers

Not actually sure right now if corner plots are also affected, this one looks fine but I'd need to double-check other examples without my messy cahnges from #171.

### CODE REVIEW

**Comment by Rodrigo-Tenorio** [comment]
Well,  these are `prior[0]` and `prior[1]`...
```
23:44 WARNING : Using saved data from PyFstat_example_data/PyFstat_example_semi_coherent_binary_search_using_MCMC/PyFstat_example_semi_coherent_binary_search_using_MCMC_saved_data.p
Hi! I'm Prior 0: -inf
/home/rodrigo/gravity/venv/pyfstat/lib/python3.8/site-packages/numpy/core/_asarray.py:83: UserWarning: Warning: converting a masked element to nan.
  return array(a, dtype, copy=False, order=order)
Hi! I'm Prior -1: -inf
Hi! I'm Prior 0: -inf
/home/rodrigo/gravity/venv/pyfstat/lib/python3.8/site-packages/numpy/core/_asarray.py:83: UserWarning: Warning: converting a masked element to nan.
  return array(a, dtype, copy=False, order=order)
Hi! I'm Prior -1: -inf
Hi! I'm Prior 0: -inf
```

**Comment by Rodrigo-Tenorio** [comment]
But that just has to do with the borders....
```
Hi! I'm Prior 1: -0.6931471805599453
Hi! I'm Prior -2: -0.6931471805599453
Hi! I'm Prior 1: -13.563967532128366
Hi! I'm Prior -2: -13.563967532128366
Hi! I'm Prior 1: -15.173405444562468
Hi! I'm Prior -2: -15.173405444562468
```
I think I really don't like the `_generic_lnprior` function at all.

**Comment by Rodrigo-Tenorio** [comment]
Ok, `_generic_lnprior` is playing the "Look at me, I've got compact suppor" game. Fine.

**Comment by Rodrigo-Tenorio** [comment]
I just solved #164 instead of #172 ... Ok, rebasing and doing another PR

**Comment by Rodrigo-Tenorio** [comment]
I'm not getting neither an error nor a walker plot, but GitHub CI's flake8 seems to catch some of them.

**Comment by dbkeitel** [comment]
Thanks for the night shift on this one, but the above example now says:
```
09:00 WARNING : Failed to plot walkers due to Error 'MCMCSearch' object has no attribute '_get_rescale_substractor_for_key'
09:00 WARNING : Failed to save walker plots due to Error local variable 'walkers_fig' referenced before assignment
```
That try-except is probably also the reason why you didn't seen an error on the intermediate commit. It seems to just be a minor typo, I'll fix it up and squash.

File: pyfstat/mcmc_based_searches.py [hunk 1]
```diff
@@ -1352,10 +1352,18 @@ def _plot_walkers(
...
-        ):
-            raise ValueError("injection_parameters is not a dictionary")
+        if injection_parameters is not None:
+            if not isinstance(injection_parameters, dict):
+                raise ValueError("injection_parameters is not a dictionary")
+            else:
+                scaled_injection_parameters = {
+                    key: (
+                        injection_parameters[key]
+                        - self._get_rescale_substractor_for_key(key)
```
Comment (dbkeitel) [inline pyfstat/mcmc_based_searches.py:1362]:
```suggestion
                        - self._get_rescale_subtractor_for_key(key)
```

**Comment by dbkeitel** [comment]
Updated and checked more examples to use this feature. Will now merge when it passes.

**Pull request merged**